packages/utils/src/object.ts

```diff
@@ -135,47 +124,38 @@ export function getWalkSource(value: any): {
 
     const event = value as unknown as SimpleEvent;
 
-    const source: {
-      [key: string]: any;
-    } = {};
-
-    // Accessing event attributes can throw (see https://github.com/getsentry/sentry-javascript/issues/768 and
-    // https://github.com/getsentry/sentry-javascript/issues/838), but accessing `type` hasn't been wrapped in a
-    // try-catch in at least two years and no one's complained, so that's likely not an issue anymore
-    source.type = event.type;
-
-    try {
-      source.target = isElement(event.target)
-        ? htmlTreeAsString(event.target)
-        : Object.prototype.toString.call(event.target);
-    } catch (_oO) {
-      source.target = '<unknown>';
-    }
-
-    try {
-      source.currentTarget = isElement(event.currentTarget)
-        ? htmlTreeAsString(event.currentTarget)
-        : Object.prototype.toString.call(event.currentTarget);
-    } catch (_oO) {
-      source.currentTarget = '<unknown>';
-    }
+    newObj = {
+      type: event.type,
+      target: serializeEventTarget(event.target),
+      currentTarget: serializeEventTarget(event.currentTarget),
+      ...getOwnProperties(event),
+    };
 
     if (typeof CustomEvent !== 'undefined' && isInstanceOf(value, CustomEvent)) {
-      source.detail = event.detail;
-    }
-
-    for (const attr in event) {
-      if (Object.prototype.hasOwnProperty.call(event, attr)) {
-        source[attr] = event[attr];
-      }
+      newObj.detail = event.detail;
     }
+  }
+  return newObj;
+}
 
-    return source;
+/** Creates a string representation of the target of an `Event` object */
+function serializeEventTarget(target: unknown): string {
+  try {
+    return isElement(target) ? htmlTreeAsString(target) : Object.prototype.toString.call(target);
+  } catch (_oO) {
+    return '<unknown>';
   }
+}
 
-  return value as {
-    [key: string]: any;
-  };
+/** Filters out all but an object's own properties */
+function getOwnProperties(obj: { [key: string]: unknown }): { [key: string]: unknown } {
```

Comment:
(Obv optional)

```suggestion
function getOwnProperties(obj: Record<string, unknown>): Record<string, unknown> {
```

Reply:
@AbhiPrasad gave me similar feedback to this for a PR I did.

Reply:
LOL - yeah. He and I default to opposite versions of this. I've actually always wanted to circumvent the whole question by creating a `PlainObject<T>` type... but that's out of scope for this PR.